app/src/features/FhirResourceTree/TreeItem.tsx

```diff
@@ -22,6 +101,12 @@ const TreeItem = ({ elementNode }: TreeItemProps): JSX.Element => {
     { skip: !isComplex || !hasExpanded || elementNode.isArray }
   );
 
+  const handleIconButtonClick = (
+    e: React.MouseEvent<HTMLButtonElement, MouseEvent>
+  ) => {
+    e.stopPropagation();
+  };
```

Comment:
Can you pre-write the 3 different kind of handlers ?

`handleAddItemClick`, `handleDeleteItemClick` & `handleAddExtensionClick`